### PR TITLE
search and filters events

### LINKED ISSUE #4: Search page and functionality
# main task

1. filter category - Drop down list for website categories
2. filter price - Filter by price dropdown list

dd

### PR DESCRIPTION
**Pull request by divluffy**
search function and filter cateogry

### CODE REVIEW

File: script/dom.js [hunk 1]
```diff
@@ -0,0 +1,3805 @@
...
+  "a star is born",
+  "a streetcar named desire",
+  "a study that measures the incidence of a disease",
+  "b symptoms",
+  "balenciaga triple s ",
+  "bank m&s",
+  "bank of the u.s ",
+  "be a m s",
+  "benz mercedes s class ",
+  "best sci fi movies",
```
Comment (Osama-you) [inline script/dom.js]: move autoComplete list to data.js file and use const instead of let

File: script/dom.js [hunk 2]
```diff
@@ -0,0 +1,3805 @@
...
+  }
+
+  if (wordAuto.length < 1) {
+    containerPopAuto.classList.remove("active");
+  }
+  test();
+});
+
+function test() {
+  const listLiAuto = document.querySelector(".list-auto-complete");
```
Comment (Osama-you) [inline script/dom.js]: Use of name and meaning for the functions

File: script/dom.js [hunk 3]
```diff
@@ -0,0 +1,3805 @@
...
+  });
+});
+
+// for delete all child
+function removeChild(parent) {
+  while (parent.firstChild) {
+    parent.removeChild(parent.firstChild);
+  }
+}
+
```
Comment (Osama-you) [inline script/dom.js]: move this function to function. js file

File: script/dom.js [hunk 4]
```diff
@@ -0,0 +1,3805 @@
...
+  divPrice.appendChild(selectPrice);
+  sectionFilter.appendChild(divPrice);
+}
+
+function filterStyleShow() {
+  let divStyleShow = document.createElement("div");
+  let styleList = document.createElement("div");
+  let styleGrid = document.createElement("div");
+  let imgList = document.createElement("img");
+  let imgGrid = document.createElement("img");
```
Comment (Osama-you) [inline script/dom.js:88]: Try to understand when you should use it const.

**Review by AhemdAboAbadi — APPROVED** [review]

**Review by Ahmad-Omar — APPROVED** [review]

**Review by AhemdAboAbadi — APPROVED** [review]

**Review by ahmadfarid-059 — APPROVED** [review]

**Pull request merged**